src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -52,8 +59,17 @@ public Operation resolveCommand(CommandContext ctx, CreateCollectionCommand comm
                 vectorSize,
                 documentLimitsConfig.maxVectorEmbeddingLength()));
       }
+      String vectorize = null;
+      if (command.options().vectorize() != null) {
+        try {
+          vectorize = objectMapper.writeValueAsString(command.options().vectorize());
+        } catch (JsonProcessingException e) {
+          // This should never happen because the object is extracted from json request
```

Comment:
which is why should probably re-throw as RuntimeException just in case it somehow happened. In theory value object serialization could fail.